### DISCUSSION TITLE
Feature: support multiple endpoints and client-side load balancer

### DISCUSSION THREAD

**Discussion by Halimao** [go-resty/resty#766] [Ideas]
It's awesome if go-resty could support setting multiple endpoints, and support client-side load balancer.

- Provide some built-in common client-side load balancer algorithm implement
- Provide a way for user to customize the load balancer

**Comment by jeevatkm** [comment]
This feature added in Resty v3 - docs: https://resty.dev/docs/load-balancer-and-service-discovery/
